### PR TITLE
Material components

### PR DESCRIPTION
**Pull request by elvi117**

### CODE REVIEW

File: modules_full/material_components/mc.info.widgets.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
+# Pre-built vs custom-built widgets
```
Comment (manzoid) [inline modules_full/material_components/mc.info.widgets.md:1]: i don't quite understand why this file is in the `Material components` module?  Isn't this more of a generic widgets thing?

**Comment by manzoid** [comment]
@elvi117 Remind me how we define the sequence of presenting these parts in the LMS? Is there some config file that does that--

**Comment by elvi117** [comment]
@manzoid yes, that's json file that define sequence

File: modules_full/material_components/mc.info.widgets.md (hunk 1 above)
Comment (elvi117) [reply modules_full/material_components/mc.info.widgets.md:1]: make sense! let's move that to "custom widgets"

File: modules_full/material_components/mc.mcq.material_design.json [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+      }
+     ],
+     "imageFit": "none"
+    },
+    {
+     "type": "radiogroup",
+     "name": "question3",
+     "title": "Material Design was introduce by: ",
```
Comment (manzoid) [inline modules_full/material_components/mc.mcq.material_design.json:94]: introduce by -> introduced by

File: modules_full/material_components/mc.mcq.material_design.json [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+      {
+       "value": "item4",
+       "text": "Microsoft"
+      }
+     ]
+    },
+    {
+     "type": "radiogroup",
+     "name": "question5",
+     "title": "Flutter forces to use Material Design in Flutters applications:",
```
Comment (manzoid) [inline modules_full/material_components/mc.mcq.material_design.json:119]: forces to use->forces you to use

Flutters->Flutter

**Review by manzoid — CHANGES_REQUESTED** [review]

File: modules_full/material_components/mc.mcq.material_design.json [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+       "value": "item2",
+       "text": "Layouts"
+      },
+      {
+       "value": "item3",
+       "text": "Animations"
+      },
+      {
+       "value": "item4",
+       "text": "Texts"
```
[the remark this answers is not in the document]
Comment (elvi117) [reply modules_full/material_components/mc.mcq.material_design.json:35]: what labels ??

File: modules_full/material_components/mc.mcq.material_design.json [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
...nhtcHRrPSJYTVAgQ29yZSA2LjAuMCI+CiAgIDxyZGY6UkRGIHhtbG5zOnJkZj0iaHR0cDovL3d3dy53My5vcmcvMTk5OS8wMi8yMi1yZGYtc3ludGF4LW5zIyI+CiAgICAgIDxyZGY6RGVzY3JpcHRpb24gcmRmOmFib3V0PSIiCiAgICAgICAgICAgIHhtbG5zOmV4aWY9Imh0dHA6Ly9ucy5hZG9iZS5jb20vZXhpZi8xLjAvIj4KICAgICAgICAgPGV4aWY6UGl4ZWxZRGltZW5zaW9uPjQwPC9leGlmOlBpeGVsWURpbWVuc2lvbj4KICAgICAgICAgPGV4aWY6UGl4ZWxYRGltZW5zaW9uPjM2PC9leGlmOlBpeGVsWERpbWVuc2lvbj4KICAgICAgICAgPGV4aWY6VXNlckNvbW1lbnQ+U2NyZWVuc2hvdDwvZXhpZjpVc2VyQ29tbWVudD4KICAgICAgPC9yZGY6RGVzY3JpcHRpb24+CiAgIDwvcmRmOlJERj4KPC94OnhtcG1ldGE+CguP6e8AAAHOSURBVFgJ7Zi/7oIwEMdPoothcTBsxjfxIRx8AUNcmJmZnEw0Dr6EOGh8DJ2IAyMbC8ZAjIl/+OUukQjS1n/4c2gTWnp3XD9825Jq6Xw+x/BDRfkhFkKRQKIZKYsCeP7ZbAaTyQQ8z4PL5cILTXyapkG73QbDMKBcvh++9OqiHg6HMBqNkoGeven1emCa5t1jLwG9C4MUtVoN1uv1HdDTizoLo+s6rFYr6Ha7lBynZLFYwHK5hHq9TrZsDBq32y0cj0fypyqcskevwWAQNxqN5Go2m3EYhvT8brcje7/fT/JZlhXnxVxzHA6HJPbK8LBCWWXwrRRFgWq1Si+oqmqqxQ7a8mIokFE9BJQHw8j3tlkIxIPBrb7f7wkiiqJUix205cVQIKPiAk2nU+7WxsFw6wdBkMTht8lxHNhsNjCfzwkoG8NgITN327daLfro8RK843NdFyqVSioFU6HT6VQoTIripsMEiuP/OZUwgW6gv3orgURyS4WkQiIFRH65hqRCIgVEfrmGpEIiBUR+5hrCkxz+mCuqYO7saRHHYgKhs9PpYFNIYeXmnqnxGDsej8G2bfB9/yNghf3Z8BG6nCTcKcuJL9wkgUQSS4VECv0B024YYgkKTfAAAAAASUVORK5CYII="
```
[the remark this answers is not in the document]
Comment (elvi117) [reply modules_full/material_components/mc.mcq.material_design.json:74]: I don't think so. base64 is the simplest way to keep images. In another case we'll need to keep these images somewhere

File: modules_full/material_components/mc.mcq.material_design.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (elvi117) [reply modules_full/material_components/mc.mcq.material_design.json:94]: removed

File: modules_full/material_components/mc.mcq.material_design.json (hunk 5 above)
[the remark this answers is not in the document]
Comment (PaulinaPogorzelska) [reply modules_full/material_components/mc.mcq.material_design.json:74]: You can just copy json paste it to [creator](https://surveyjs.io/create-survey) in JSON editor tab and click preview tab to see how quiz will look @manzoid

File: modules_full/material_components/mc.code.cats_login.md [hunk 6]
```diff
@@ -0,0 +1,31 @@
+# Login page
+
+Try to reproduce followed page:
```
Comment (manzoid) [inline modules_full/material_components/mc.code.cats_login.md:3]: the following page

File: modules_full/material_components/mc.code.cats_login.md [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+
+<img src="https://i.imgur.com/ZtfaapT.png" height="600"/>
+
+
+#### #1: Look carefully, as you can see each button has additional subtle aspects, such as:
+  * slight rounding of the corners
+  * a slight "drop shadow". 
+
+Use Material widget called [Card](https://api.flutter.dev/flutter/material/Card-class.html) to for this effect. As you can find in the official documentation of the `Card` widget, there is another widget which really likes to be Card's child widget: [ListTile](https://api.flutter.dev/flutter/material/ListTile-class.html).
```
Comment (manzoid) [inline modules_full/material_components/mc.code.cats_login.md:12]: to for -> for

File: modules_full/material_components/mc.code.cats_login.md [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+
+`ListTile` gives us the `leading` and `title` properties for pretty rendering.
+
+#### #2: All images you can find in `images` directory.
+
+Remember to load them.
+
+#### #3: You can use hexadecimal colors in Flutter.
+
+To do so, use the syntax `Color(0x...)` where '...' is colors in HEX. Here are the colors necessary to do this exercise:
```
Comment (manzoid) [inline modules_full/material_components/mc.code.cats_login.md:22]: HEX -> hexadecimal

File: modules_full/material_components/mc.code.summarist_book_details.md [hunk 9]
```diff
@@ -0,0 +1,10 @@
+# Book details
+
+Reproduce followed screen.
```
Comment (manzoid) [inline modules_full/material_components/mc.code.summarist_book_details.md:3]: followed -> the following 

(please learn these bits of English @elvi117 -- i know i have corrected this exact phrasing in past projects! :)

File: modules_full/material_components/mc.code.summarist_book_details.md [hunk 10]
```diff
@@ -0,0 +1,10 @@
...
+
+<img src="https://i.imgur.com/KuEiOHH.mp4" width="260" loading="lazy"/>
+
+1. Page can be higher than device screen, use [SingleChildScrollView](https://api.flutter.dev/flutter/widgets/SingleChildScrollView-class.html) widget to make it scrollable.
```
Comment (manzoid) [inline modules_full/material_components/mc.code.summarist_book_details.md:7]: than the device screen

File: modules_full/material_components/mc.code.summarist_book_details.md [hunk 11]
```diff
@@ -0,0 +1,10 @@
...
+2. You don't need to copy all texts, just use text generator called [Lorem Ipsum](https://www.lipsum.com/).
```
Comment (manzoid) [inline modules_full/material_components/mc.code.summarist_book_details.md:8]: all texts -> all the text

"Texts" as a plural usually means books or manuscripts, etc

Comment (manzoid) [inline modules_full/material_components/mc.code.summarist_book_details.md:8]: use the text generator

File: modules_full/material_components/mc.info.design_system.md [hunk 12]
```diff
@@ -0,0 +1,53 @@
...
+
+#### Example: standardized icons
+
+<img src=https://i.imgur.com/zigWw3o.png height=400>
+<img src=https://i.imgur.com/1asXIr0.png height=400>
+<img src=https://i.imgur.com/jczNCBL.png height=400>
+
+In the examples above, you can find common icons. You've seen these thousands of times, yourself.
+
+Why do you know that "magnifying glass" icon means _search_ or three dots ('...') means _more options_ ?
```
Comment (manzoid) [inline modules_full/material_components/mc.info.design_system.md:35]: that the

File: modules_full/material_components/mc.mcq.material_design.json [hunk 13]
```diff
@@ -0,0 +1,112 @@
...
+       "value": "item1",
+       "text": "Notification"
+      },
+      {
+       "value": "item2",
+       "text": "Settings"
+      },
+      {
+       "value": "item3",
+       "text": "Account information"
```
Comment (manzoid) [inline modules_full/material_components/mc.mcq.material_design.json:57]: this one seems potentially confusing. "Settings" is often short for "account settings"…

File: modules_full/material_components/mc.mcq.material_design.json [hunk 14]
```diff
@@ -0,0 +1,112 @@
...
+      {
+       "value": "item4",
+       "text": "Search"
+      }
+     ]
+    },
+    {
+     "type": "imagepicker",
+     "name": "question4",
+     "title": "Search features in Material Design are represented by:",
```
Comment (manzoid) [inline modules_full/material_components/mc.mcq.material_design.json:68]: are you sure this isn't redundant with the gear icon question? 

how about a question that checks if they understood the developer benefits of design systems, instead?

**Review by manzoid — CHANGES_REQUESTED** [review]

File: modules_full/material_components/mc.info.widgets.md (hunk 1 above)
Comment (elvi117) [reply modules_full/material_components/mc.info.widgets.md]: removed

File: modules_full/material_components/mc.mcq.material_design.json (hunk 4 above)
[the remark this answers is not in the document]
Comment (elvi117) [reply modules_full/material_components/mc.mcq.material_design.json:35]: okay ! I see ! 😄 
I thought about text (copy) like "lorem ipsum" etc... 
how can I call that?

File: modules_full/material_components/mc.mcq.material_design.json (hunk 14 above)
Comment (elvi117) [reply modules_full/material_components/mc.mcq.material_design.json:68]: do you have any idea ?

File: modules_full/material_components/mc.mcq.material_design.json (hunk 13 above)
Comment (elvi117) [reply modules_full/material_components/mc.mcq.material_design.json:57]: changed to "news"

**Comment by PaulinaPogorzelska** [comment]
> when these are fixed please merge and let's move to staging this today @elvi117 @PaulinaPogorzelska

After Łukasz will provide curriculum i will go to [task with updating curriculum upload script](https://tpa-lms.atlassian.net/browse/TPA-348) and after it upload it to staging. 
Today I'm closing backend part for mcq's so we can can schedule this for Monday. @manzoid

**Pull request merged**